src/adaptors/mod.rs

```diff
@@ -761,6 +761,26 @@ macro_rules! impl_tuple_combination {
                 let n = self.iter.count();
                 checked_binomial(n, K).unwrap() + self.c.count()
             }
+
+            fn fold<B, F>(self, mut init: B, mut f: F) -> B
+            where
+                F: FnMut(B, Self::Item) -> B,
+            {
+                let Self { c, item, mut iter } = self;
+                init = c
+                    .map(|($($X),*,)| {
```

Comment:
Do we need `$($X),*,`? Could we use the simpler `$($X,)*`?

Reply:
In `impl_tuple_combination!(Tuple2Combination Tuple1Combination; a);`, `($($X,)*)` expands to `(a)` which is not the wanted tuple: `(a,)`.

Reply:
Maybe I'm wrong, but we already use `$($X,)*` in `size_hint`. I think it expands to `(a,)`.

Reply:
To be sure I removed the comma and had the compiler error I thought I would have.
In `size_hint`, commas are only given to `count_ident!` macro, which does not want a tuple so it's not `(... ,)` wrapped.

Reply:
Honestly, I think the `count_ident` macro should work without any comma too. It does not need those commas.

Reply:
`$($X,)*` would be simpler and should have the same result.

Reply:
There were four places where `($($X,)*)` is better than `($($X),*,)`, committed.